specs/Standards/Wallets/InjectedWallets.md

```diff
@@ -0,0 +1,369 @@
+
+- It future-proofs the API even if wallets (such as MetaMask) only support a single "active" account.
+- Other blockchains such as [Ethereum](https://docs.metamask.io/guide/rpc-api.html#eth-requestaccounts) implement this model.
+- Access to multiple accounts allow dApps more freedom to improve UX as users can seamlessly switch between accounts.
+- Aligns with WalletConnect via the [Bridge Wallet Standard](./BridgeWallets.md).
+
+### Store key pairs of FunctionCall access keys in dApp vs. wallet
+
+NEAR's unique concept of `FunctionCall` access keys makes it possible for dApps to execute transactions without having to prompt the user each time. This limited access means they can be shared with relatively low risk, provided gas allowance and method names are appropriately defined.
+
+There has been mixed views on exactly who should store these key pairs as there are trade-off to be considered with either option:
+
+- Storing key pairs in the wallet means users have a contextual view of which dApps have limited access to their accounts. However, this can be solved through improvements to the `view_access_key` endpoints to include a description and/or url that helps users make a more informed decision on whether they want to remove it.
+- Although risk is limited with `FunctionCall` access keys, storing key pairs in the dApp means a level of trust is required to ensure they're kept securely to avoid being compromised.
+- Signing transactions that match the `FunctionCall` access key stored in the dApp means we don't require connection to a wallet. This is particularly useful for wallets such as Ledger and WalletConnect where it isn't always available. The drawback to this approach is the dApp must handle much of the logic found already in the wallet such as storing key pairs and matching applicable access keys during signing.
+
+Although we had some technical challenges, the decision to store the key pairs in the dApp means users own them and gas-only intensive `FunctionCall` dApps work seamlessly with wallets that aren't always available. The expectation for `near-api-js` to become wallet-agnostic will reduce the complexity shifted onto the dApp as it can abstract away the logic required for handling the key pairs and only defer to the wallet when further permission is needed.
+
+A side effect to this approach (coupled with multiple accounts) means we must generate the key pairs ahead of time and pass each public key to the `signIn` method as a list of `accounts`. The introduction of the `connect` and `disconnect` methods allows dApps to request visibility for a subset of accounts imported in the wallet to help populate this list and sign transactions.
+
+## Specification
+
+Injected wallets are browser extensions that implement the `Wallet` API (see below) on the `window` object. To avoid namespace collisions and easily detect when they're available, wallets will mount under their own key within `window.near` (e.g. `window.near.sender`).
```

Comment:
I think this may open up the door to some potential issues. Sender is already using `window.near` to host their wallet API- but besides that, to assign deeply (2 levels) into the window object might make people inadvertently remove previous assignments:

```ts
window.near = {}; // This will remove all other implementations
window.near.sender = { ...senderApi };
```

A way to get around it would be:

```ts
if (window.near == null) {
  window.near = {};
}

window.near.sender = { ...senderApi };
```

Hopefully people implement their extensions like this- but having this kind of deep assignment does open to door to people getting it wrong and inadvertently voiding others' APIs.

Reply:
:+1: good point -- I've added some text to warn wallet authors against overwriting the object